### PR TITLE
Implement progress callback so node domains are able to notify the Brackets before they finish

### PR DESCRIPTION
**Pull request by zaggino**
Related PR: https://github.com/adobe/brackets-shell/pull/509
Related Issue: https://github.com/adobe/brackets/issues/7356

Please review @nethip , I'd like to get this into Brackets as this would help to improve user experience on some of my work I'm doing with extensions.

### CODE REVIEW

**Comment by zaggino** [comment]
If we don't want to introduce `progressCallback` as a new argument to node functions,  we can make a call `callback(null, null, "progress")` to invoke a progress and not resolve the promise checking for `arguments.length === 3`

**Comment by nethip** [comment]
@zaggino Sure! I will look this!

**Comment by zaggino** [comment]
The more I'm thinking about this, the more `callback(null, null, "progress")` version makes sense, but I'll wait for some opinions before fixing this PR.

**Comment by nethip** [comment]
@zaggino Sorry. I could not find some time to go through this PR as I was caught up with lot of other stuff. I will for sure take a look at this PR next week.

**Comment by zaggino** [comment]
So I'm guessing nobody found time to review this for 1.3 ...

**Comment by nethip** [comment]
Yes @zaggino That is right. Sorry for not being able to dedicate some time to this PR in this release.

**Comment by zaggino** [comment]
Can we get someone to merge this into 1.4?

**Comment by pvaishnav** [comment]
Picking for review.

**Comment by js0511** [comment]
NAVER - http://www.naver.com/
--------------------------------------------

oiup2006@naver.com 님께 보내신 메일 <Re: [brackets] Implement progress callback so node domains are able to notify the Brackets before they finish (#10761)> 이 다음과 같은 이유로 전송 실패했습니다.

--------------------------------------------

받는 사람이 회원님의 메일을 수신차단 하였습니다. 


--------------------------------------------

File: test/spec/NodeConnection-test.js [hunk 1]
```diff
@@ -192,6 +192,42 @@ define(function (require, exports, module) {
             });
         });
         
+        it("should receive progress events from asynchronous commands", function () {
```
Comment (abose) [inline test/spec/NodeConnection-test.js]: @zaggino I ran the Live preview test for `Node connection` and the test is timing out at `should receive progress events from asynchronous commands` [win 8.1]. Any idea why?

Comment (prafulVaishnav) [inline test/spec/NodeConnection-test.js]: @abose Have you taken shell changes also? https://github.com/adobe/brackets-shell/pull/509

Comment (abose) [inline test/spec/NodeConnection-test.js]: oh! i didn't that explains. Did you verify this change?

**Comment by abose** [comment]
Verified, merging. 
The corresponding PR adobe/brackets-shell#509 in brackets shell has been merged. 
Our linux builds should pick up this change on brackets shell as we build from the older CEF branch.

**Comment by nethip** [comment]
This is going to make @zaggino to see this once he is back from vacation :smile:

**Comment by zaggino** [comment]
Thanks a lot guys for merging this :+1:

**Pull request merged**